### PR TITLE
Add global log level through conf file

### PR DESCRIPTION
**Pull request by efrgmbh**
With a *global* section in the `/etc/zlog.conf` one can define a level to fine-tune output without touching the source code.

E.g. with levels like `fatal > error > warn > notice > info > debug > trace`
If one sets the level to `info` the debug and trace logs wouldn't be emitted.

```
[global]

strict init = true
log level = notice
```

### CODE REVIEW

File: src/category.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ void zlog_category_rollback_rules(zlog_category_t * a_category);
 int zlog_category_output(zlog_category_t * a_category, zlog_thread_t * a_thread);
 
 #define zlog_category_needless_level(a_category, lv) \
-        !((a_category->level_bitmap[lv/8] >> (7 - lv % 8)) & 0x01)
+        a_category && (zlog_env_conf->level > lv || !((a_category->level_bitmap[lv/8] >> (7 - lv % 8)) & 0x01))
```
Comment (deemic) [inline src/category.h]: It has conflict with your last PR

Comment (lnksz) [reply src/category.h]: rebased.

Comment (efrgmbh) [reply src/category.h]: rebased

**Comment by deemar** [comment]
I think global log level should be configured dynamically in run time. This configuration will make user confused.

**Comment by lnksz** [comment]
But a config file *is* having its effect at runtime...

See log4j's `<ThresholdFilter level="trace"/>`

**Comment by liusuchao** [comment]
感谢，您的更新。zlog非常好用！支持您！

**Pull request merged**